jaxrs/src/main/java/io/narayana/lra/filter/ServerLRAFilter.java

```diff
@@ -406,7 +413,19 @@ public void filter(ContainerRequestContext containerRequestContext) {
                     // store the registration link in case the participant wants to associate data with the enlistment in the LRA
                     containerRequestContext.setProperty(PARTICIPANT_LINK_PROP, compensatorLink);
 
-                    recoveryUrl = getLRAClient().enlistCompensator(lraId, timeLimit, compensatorLink, previousParticipantData);
+                    // The coordinator needs to hold a lock to enlist an participant. This lock is only waited on for a small amount of
+                    // time (potentially even zero time). This means if multiple participants try to enlist at the same time, enlistment
+                    // may fail. We therefore re-try a configurable amount of times.
+                    for (int i =0;; i++) {
```

Comment:
@arjantijms I think you are missing the condition "i < ENLIST_PARTICIPANT_CLIENT_MAX_RETRY" here

Reply:
Sorry I missed it. I see. Could I ask you to log a debug message in the catch block please?
I understand that it could occur frequently when having multiple enlistments in parallel, but I think it might help. I would log something like 
_"The service is unavailable, failing in enlisting the compensator for " + lraId + ", ENLIST_PARTICIPANT_CLIENT_MAX_RETRY set to  " + enlistMaxRetries."_

Reply:
That sounds good!